app/uk/gov/hmrc/gform/formtemplate/DestinationsValidator.scala

```diff
@@ -36,15 +39,25 @@ object DestinationsValidator {
       duplicates.isEmpty.validationResult(someDestinationIdsAreUsedMoreThanOnce(duplicates))
   }
 
+  def validateNoGroupInDeclaration(destinations: Destinations): ValidationResult = destinations match {
+    case _: Destinations.DestinationPrint => Valid
+    case destinationList: Destinations.DestinationList =>
+      val groupComponentIds = extractGroupComponentIds(destinationList.declarationSection.fields)
+      groupComponentIds.isEmpty.validationResult(groupComponentInDeclaration(groupComponentIds))
+  }
+
+  def extractGroupComponentIds(fcs: List[FormComponent]): List[FormComponentId] = fcs
+    .map(fc => (fc.id, fc.`type`))
+    .collect { case (fcId, Group(gFields, _, _, _, _)) =>
+      fcId :: extractGroupComponentIds(gFields)
+    }
+    .flatten
```

Comment:
We can make this simpler, by returning only `Option[FormComponentId]`. 

Also recursion can be eliminated, since outer `Group` will be always detected.

Also use existing `IsGroup` extractor rather than custom `map(...).collect(...)` combo. My guess would be to implemented it somehow like `fcs.collectFirst { case g @ IsGroup(_)) => g.id }`